build-tools/automation/azure-pipelines.yaml

```diff
@@ -52,665 +56,597 @@ variables:
-        testResultsFile: TestResult-XASdkTests-Linux-$(XA.Build.Configuration).xml
+        - template: /build-tools/automation/yaml-templates/fail-on-issue.yaml@self
 
-    - template: yaml-templates/run-nunit-tests.yaml
-      parameters:
-        testRunTitle: Xamarin.Android.Build.Tests - Linux AndroidDependenciesTests
-        testAssembly: $(System.DefaultWorkingDirectory)/bin/Test$(XA.Build.Configuration)/$(DotNetStableTargetFramework)/Xamarin.Android.Build.Tests.dll
-        dotNetTestExtraArgs: --filter "Name = AndroidDependenciesTests"
-        testResultsFile: TestResult-AndroidDependenciesTests-Linux-$(XA.Build.Configuration).xml
-
-    - task: ShellScript@2
-      displayName: Test dotnet-local.sh
-      inputs:
-        scriptPath: dotnet-local.sh
-        args: build samples/HelloWorld/HelloWorld/HelloWorld.DotNet.csproj
-
-    - template: yaml-templates/upload-results.yaml
+    - template: /build-tools/automation/yaml-templates/stage-msbuild-tests.yaml@self
+
+    - template: /build-tools/automation/yaml-templates/stage-msbuild-emulator-tests.yaml@self
       parameters:
-        configuration: $(XA.Build.Configuration)
-        artifactName: Test Results - MSBuild - Linux
+        usesCleanImages: ${{ parameters.macTestAgentsUseCleanImages }}
+
+    - stage: maui_tests
+      displayName: MAUI Tests
+      dependsOn: mac_build
+      condition: and(eq(dependencies.mac_build.result, 'Succeeded'), eq(variables['RunMAUITestJob'], 'true'))
+      jobs:
+      # Check - "Xamarin.Android (MAUI Tests MAUI Integration)"
+      - job: maui_tests_integration
+        displayName: MAUI Integration
+        pool:
+          name: MAUI-1ESPT
+          image: $(WindowsPoolImage1ESPT)
+          os: windows
+        timeoutInMinutes: 180
+        workspace:
+          clean: all
+        variables:
+          BuildVersion: $(Build.BuildId)
+        steps:
+        - checkout: maui
+          clean: true
+          submodules: recursive
+          path: s/maui
+          persistCredentials: true
+
+        - template: /build-tools/automation/yaml-templates/setup-test-environment.yaml@self
+          parameters:
+            xaSourcePath: $(Build.SourcesDirectory)/xamarin-android
+            androidSdkPlatforms: $(DefaultTestSdkPlatforms)
+
+        - task: NuGetAuthenticate@1
+          displayName: authenticate with azure artifacts
+          inputs:
+            forceReinstallCredentialProvider: true
 
-    - template: yaml-templates/fail-on-issue.yaml
+        - script: |
+            echo ##vso[task.setvariable variable=JI_JAVA_HOME]%JAVA_HOME_17_X64%
+            echo ##vso[task.setvariable variable=JAVA_HOME]%JAVA_HOME_17_X64%
+          displayName: set JI_JAVA_HOME, JAVA_HOME to $(JAVA_HOME_17_X64)
 
-- template: yaml-templates/stage-msbuild-tests.yaml
+        - script: echo "##vso[task.prependpath]C:\Windows\System32\WindowsPowerShell\v1.0\"
+          displayName: add powershell to path
```

Comment:
This seems like it is in my `%PATH%` by default already. Does one of the 1ES build agents not have this?

Reply:
For some reason the MAUI integration test job was failing on the new pool until I added this:

> Restored D:\a\_work\1\s\maui\src\DotNet\DotNet.csproj (in 7.3 sec).
  Downloading from "[https://dot.net/v1/dotnet-install.ps1"](https://dot.net/v1/dotnet-install.ps1%22) to "D:\a\_work\1\s\maui\bin\dotnet-install.ps1" (70,702 bytes).
  'powershell' is not recognized as an internal or external command, operable program or batch file.